modules/api_loader.lua

```diff
@@ -9,73 +9,148 @@ local function get_name(xml_element)
     end
     return name
 end
+ 
 
-local function loadEnums(api, dest)
-  local enums = api:xpath("//interface/enum")
-  for _, e in ipairs(enums) do
-    local enum = { }
-    local i = 1
-    for _, item in ipairs(e:children("element")) do
-      enum[item:attr("name")] = i
-      i = i + 1
+local function LoadResultCodes( param )
+  local resultCodes ={}
+  local i = 1
+  for _, item in ipairs(param:children("element")) do
+     local name = item:attr("name")
+     resultCodes[i]=name
+     i=i + 1
     end
-    dest.enum[get_name(e)] = enum
+  return resultCodes
+end
+
+local function LoadParamsInFunction(param, interface)
+  local name = param:attr("name")
+  local p_type = param:attr("type")
+  local minlength = tonumber(param:attr("minlength"))
+  local maxlength = tonumber(param:attr("maxlength"))
+  local minsize =  tonumber(param:attr("minsize"))
+  local maxsize = tonumber(param:attr("maxsize"))
+  local minvalue = tonumber(param:attr("minvalue"))
+  local maxvalue = tonumber(param:attr("maxvalue"))
+  local mandatory = param:attr("mandatory")
+  local array = param:attr("array")
+
+  if mandatory == nil then 
+    mandatory = true
```

Comment:
Will it more readable to concatenate attributes loading and checking?
like following
```
 local mandatory = param:attr("mandatory") then true;
```